File: Csharp/They say that only the name is long.cs
https://www.codewars.com/kata/574b1916a3ebd6e4fa0012e7/train/csharp

using System.Linq;
using System;
public class Kata {
  public static bool IsOpposite(string s1, string s2)
   => string.IsNullOrEmpty(s1) ? false :   
       s1.SequenceEqual( s2.Select(e=>  Char.IsLower(e) ? Char.ToUpper(e) : Char.ToLower(e))); 
}

File: javascript/They say that only the name is long.js
// https://www.codewars.com/kata/574b1916a3ebd6e4fa0012e7/train/javascript

function isOpposite(s1,s2){
    return s1.split('')
      .map(c => c === c.toUpperCase() ? c.toLowerCase() : c.toUpperCase())
      .join('') === s2 && s1 !== '';
    
  }